sklearn/decomposition/_lda.py

```diff
@@ -85,15 +85,19 @@ def _update_doc_distribution(
     n_topics = exp_topic_word_distr.shape[0]
 
     if random_state:
-        doc_topic_distr = random_state.gamma(100.0, 0.01, (n_samples, n_topics))
+        doc_topic_distr = random_state.gamma(100.0, 0.01, (n_samples, n_topics)).astype(
+            X.dtype
+        )
```

Comment:
We can save a memory copy for `float64` by:

```suggestion
        doc_topic_distr = random_state.gamma(100.0, 0.01, (n_samples, n_topics)).astype(
            X.dtype, copy=False
        )
```

Same comment from your other PR https://github.com/scikit-learn/scikit-learn/pull/22114/files#r777245455 . In this case, it would be nice to use [Generator.standard_gamma](https://numpy.org/devdocs/reference/random/generated/numpy.random.Generator.standard_gamma.html), but we would need to support generators first.

Reply:
Thank you for the explanation.
I have specified `copy=False`